### ISSUE TITLE
[meta] Use FTS in places

### ISSUE THREAD

**Issue by data-sync-user** [mozilla/application-services#2924]
**TL;DR Why:** Implement FTS in places for iOS to minimize perf. regression with the places rust component upgrade.  

Currently we search history using a port of desktop's autocomplete match. The single largest noted deficiency of desktop places was the lack of FTS, and the main reason we reimplemented AUTOCOMPLETE_MATCH was because we felt that there must be some reason places didn't implement it, using the notes in [bug 1340487](https://bugzilla.mozilla.org/show_bug.cgi?id=1340487) as guidance.

However, having done that port, the naive (unicode61) FTS solution is no worse than what we do now (and better in many ways, including handling of unicode text and performance), so we should do it when we can (both for those benefits, and to gain experience with sqlite's FTS features).

┆Issue is synchronized with this [Jira Task](https://jira.mozilla.com/browse/SYNC-948)
┆Sprint: Backlog

**Comment by data-sync-user** [comment]
➤ Ana Medinac commented:

Later. Revisit this once iOS wants to upgrade to the History rust component in order to minimize perf. regression with the upgrade.

**Comment by data-sync-user** [comment]
➤ Janet Dragojevic commented:

closing  - duplicate of SYNC-667
